### PR TITLE
Add a functionality for publishing to GitHub Pages

### PR DESCRIPTION
**Pull request by jiri-one**
PR to fix issue #267

### CODE REVIEW

File: freezeyt/filesaver.py [hunk 1]
```diff
@@ -15,6 +17,21 @@ class FileSaver(Saver):
...
+        Function parameters are:
+        function: function which raised the exception,
+        path: path name passed to function,
+        excinfo: exception information returned by sys.exc_info()
+        """
+        if not os.access(path, os.W_OK):
+            os.chmod(path, stat.S_IWRITE)
+            function(path)
+        else:
+            raise
```
Comment (encukou) [inline freezeyt/filesaver.py:33]:
```suggestion
            raise excinfo[1]
```

File: freezeyt/filesaver.py [hunk 2]
```diff
@@ -15,6 +17,21 @@ class FileSaver(Saver):
...
+    def add_write_flag(function, path, excinfo):
+        """A function that adds a write attribute/flag for a path where such an attribute is missing. This function is not necessary on Linux, but on Windows, attempting to delete a file where such an attribute is missing will raise an exception.
+        
+        Function parameters are:
+        function: function which raised the exception,
+        path: path name passed to function,
+        excinfo: exception information returned by sys.exc_info()
+        """
+        if not os.access(path, os.W_OK):
+            os.chmod(path, stat.S_IWRITE)
```
Comment (encukou) [inline freezeyt/filesaver.py:30]:
```suggestion
            os.chmod(path, os.stat(path).st_mode | stat.S_IWRITE)
```

File: tests/test_gh_pages.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+              "output": str(output_dir)}
+    freeze(app, config)
+    assert (output_dir / ".git").exists() # the .git directory has to exist
+    assert (output_dir / ".git").is_dir() # the .git directory has to be directory
+    assert (output_dir / ".git/refs/heads/gh-pages").exists() # the gh-pages has to be git head branch
+    head_name = check_output(["git", "name-rev", "--name-only", "HEAD"], cwd=output_dir).decode().strip() # get name of HEAD branch
+    assert "gh-pages" == head_name # the gh-pages has to be git head branch
+    last_commit_info = check_output(["git", "rev-list", "--oneline", "HEAD"], cwd=output_dir).decode().strip()
+    last_commit_hash = last_commit_info[:7]
+    last_commit_text = last_commit_info[8:]
```
Comment (encukou) [inline tests/test_gh_pages.py:40]:
```suggestion
    last_commit_hash, last_commit_text = last_commit_info.split(' ', 1)
```

Comment (encukou) [inline tests/test_gh_pages.py:40]:
```suggestion
    last_commit_hash, last_commit_text = last_commit_info.split(' ', 1)
```

**Comment by jiri-one** [comment]
What is missing is documentation, including instructions on how to pull files from the git repository you created. And I don't know whether to add some message to the final listing that the git repository gh-pages was successfully created.

File: tests/test_gh_pages.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    freeze(app, config)
+    freeze(app, config)
+    expected_files = ['.nojekyll', 'CNAME', 'index.html', 'second_page.html']
+    number_of_files = 0
+    for file in output_dir.iterdir():
+        # the .git directory is not created on all systems/configurations
+        if file.name != ".git": 
+            assert file.name in expected_files
+            number_of_files += 1
+    assert number_of_files == len(expected_files)
```
Comment (encukou) [inline tests/test_gh_pages.py:88]:
```suggestion
    expected_files = {'.nojekyll', 'CNAME', 'index.html', 'second_page.html'}
    got_files = {name for name in output_dir.iterdir() if name != ".git"}
    assert got_files == expected_files
```

**Review by encukou — APPROVED** [review]

**Pull request merged**